common/enforce/assertions_test.go

```diff
@@ -0,0 +1,182 @@
+package enforce
+
+import "testing"
+
+func TestTrue(t *testing.T) {
+	True(true, "This should not panic")
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("Expected panic, but did not panic")
+		}
+	}()
+
+	True(false, "This should panic, and the deferred function will catch it")
+}
+
+func TestFalse(t *testing.T) {
+	False(false, "This should not panic")
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("Expected panic, but did not panic")
+		}
+	}()
+
+	False(true, "This should panic, and the deferred function will catch it")
+}
+
+func TestEquals(t *testing.T) {
+	Equals(1, 1, "This should not panic")
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("Expected panic, but did not panic")
+		}
+	}()
+
+	Equals(1, 2, "This should panic, and the deferred function will catch it")
+}
+
+func TestNotEquals(t *testing.T) {
+	NotEquals(1, 2, "This should not panic")
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("Expected panic, but did not panic")
+		}
+	}()
+
+	NotEquals(1, 1, "This should panic, and the deferred function will catch it")
+}
+
+func TestGreaterThan(t *testing.T) {
```

Comment:
consider adding a test case for ==, which should panic for `GreaterThan`

Reply:
added